account_payment_mode/models/account_payment_mode.py

```diff
@@ -0,0 +1,123 @@
+    _description = 'Payment Modes'
+    _order = 'name'
+
+    name = fields.Char(string='Name', required=True, translate=True)
+    company_id = fields.Many2one(
+        'res.company', string='Company', required=True, ondelete='restrict',
+        default=lambda self: self.env['res.company']._company_default_get(
+            'account.payment.mode'))
+    bank_account_link = fields.Selection([
+        ('fixed', 'Fixed'),
+        ('variable', 'Variable'),
+        ], string='Link to Bank Account', required=True,
+        help="For payment modes that are always attached to the same bank "
+        "account of your company (such as wire transfer from customers or "
+        "SEPA direct debit from suppliers), select "
+        "'Fixed'. For payment modes that are not always attached to the same "
+        "bank account (such as SEPA Direct debit for customers, wire transfer "
+        "to suppliers), you should select 'Variable', which means that you "
+        "will select the bank account on the payment order. If your company "
+        "only has one bank account, you should always select 'Fixed'.")
+    fixed_journal_id = fields.Many2one(
+        'account.journal', string='Fixed Bank Journal',
+        domain=[('type', '=', 'bank')], ondelete='restrict')
+    # I need to use the old definition, because I have 2 M2M fields
+    # pointing to account.journal
+    variable_journal_ids = fields.Many2many(
+        comodel_name='account.journal',
+        relation='account_payment_mode_variable_journal_rel',
+        column1='payment_mode_id', column2='journal_id',
+        string='Allowed Bank Journals')
+    payment_method_id = fields.Many2one(
+        'account.payment.method', string='Payment Method', required=True,
+        ondelete='restrict')  # equivalent v8 field : type
+    payment_type = fields.Selection(
+        related='payment_method_id.payment_type', readonly=True, store=True,
+        string="Payment Type")
+    payment_method_code = fields.Char(
+        related='payment_method_id.code', readonly=True, store=True,
+        string='Payment Method Code')
+    active = fields.Boolean(string='Active', default=True)
+    # I dropped sale_ok and purchase_ok fields, because it is replaced by
+    # payment_type = 'inbound' or 'outbound'
+    # In fact, with the new v9 datamodel, you MUST create 2 payment modes
+    # for wire transfer : one for wire transfer from your customers (inbound)
+    # and one for wire transfer to your suppliers (outbound)
+    note = fields.Text(string="Note", translate=True)
+
+    @api.onchange('company_id')
+    def _onchange_company_id(self):
+        for mode in self:
+            mode.variable_journal_ids = False
+            mode.fixed_journal_id = False
+
+    @api.constrains(
+        'bank_account_link', 'fixed_journal_id', 'payment_method_id')
+    def bank_account_link_constrains(self):
+        for mode in self:
+            if mode.bank_account_link == 'fixed':
```

Comment:
I would put this if up
´for mode in self.filtered(lambda x: x.bank_account_link == 'fixed'):´